### DISCUSSION TITLE
Is it possible to automatically assign authors on a space?

### DISCUSSION THREAD

**Discussion by asikranjm** [snapshot-labs/snapshot#2775] [Q&A]
I would like to know if there is an automatic way to assign authors on a space. I mean not manually. It's possible?

**Comment by samwhosung** [comment]
@asikranjm Under which conditions would you like to add Authors automatically?

**Reply by asikranjm** [reply]
Thanks @samuveth. It could be said, when the user complies with some method of a smart contract. For example, if you have a certain amount of tokens in Staking for a certain time, in that case we let you create proposals. This would be programmed in a method on a smart contract that returns if it meets the condition.

**Reply by samwhosung** [reply]
@asikranjm you could achieve this in the space settings with "Validation". If none of the available validations work for you, you can make a PR to add your own. https://github.com/snapshot-labs/snapshot.js/tree/master/src/validations

**Reply by asikranjm** [reply]
Thanks again @samuveth. I understand that it has nothing to do with the strategy used for voters, right?

**Reply by samwhosung** [reply]
If you use the Threshold setting then it uses strategies. With a custom validation you could deicide that yourself :)

**Reply by nazhG** [reply]
We're try to use a `contract-call` strategy in the Proposal validation but we can't setup, something like this
```
{
  "strategies": [
    {
      "name": "contract-call",
      "params": {
       {
        "symbol": "LP",
        "address": "0x67E6F294919DD283cA6A73f73b29DC35DddB8D50",
        "decimals": 18,
        "methodABI": {
            "name": "getRole",
            "type": "function",
            "inputs": [
            {
                "name": "_account",
                "type": "address",
                "internalType": "address"
            }
            ],
            "outputs": [
            {
                "name": "",
                "type": "uint256",
                "internalType": "uint256"
            }
            ],
            "stateMutability": "view"
            }
        }
      }
    }
  ]
}
```
Is this possible? Can we see an example if that is the case?
